### DISCUSSION TITLE
File Errors?

### DISCUSSION THREAD

**Discussion by katieliuu** [CS342/.github#9] [Technical Questions] [answered]
### In what area do you have a technical challenge?

Xcode, Simulator & Previews

### Description

<img width="892" alt="Screenshot 2023-01-21 at 1 11 42 AM" src="https://user-images.githubusercontent.com/122421205/213860367-d0280259-076e-4836-ba1f-3087980f8116.png">

Hi! I had this weird error in my Xcode file while I was working on my application. How might I be able to resolve this? Thanks so much!

### Reproduction

I added nested files and folders in my Resources folder.

### Expected behavior

There should not have been an error.

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by PSchmiedmayer** [comment, marked as answer]
@katieliuu Thank you for posting your question!

The error is probably related to your `Package.swift` configuration. E.g., my `Target` configuration looks like this:
```swift
.target(
    name: "PaulSchmiedmayer",
    dependencies: [
        .target(name: "InstructorViews")
    ],
    exclude: [
        "Resources/PaulSchmiedmayer.jpeg.license",
        "Resources/PaulSchmiedmayerBio.md.license"
    ],
    resources: [
        .process("Resources")
    ]
),
```

The important part to focus on here is the `exclude` `Array` of `String`s. This defines what resources (defined in the `resources` property to just process the complete `"Resources"` subfolder) should be excluded.
As @vishnuravi has pointed out in #8, we want all source code to conform to the [REUSE Specification](https://reuse.software/spec/) that defines a standardized method for declaring copyright and licensing in software projects. Files like an image can not easily include a header that is required to conform to the REUSE spec and therefore, we had a separate file prefixed with `.license` in the folder to ensure that all files are properly licensed. 

The error message you encountered is due to the fact that you still have defined that there should be a, e.g., a `KatieLiu.jpg.license` file in the `Resources` folder of your target while there is no file with that name present if you inspect your resources folder. If you have an image (e.g., `KatieLiu.jpg`) add the `KatieLiu.jpg.license` file with the appropriate license or otherwise remove the `exclude` `String` instances that are no longer needed.

**Reply by katieliuu** [reply]
Thanks so much for the detailed reply!! I did add those two files to my Resources folder - but for some reason the error still shows? I've attached an image below. 
<img width="398" alt="Screenshot 2023-01-21 at 10 35 05 PM" src="https://user-images.githubusercontent.com/122421205/213903884-2c8b6c6b-c482-4ab0-b48e-7555c82ae74e.png">

**Reply by PSchmiedmayer** [reply]
@katieliuu Can you share your Swift Package Manifest file with us in a comment? GitHub allows you to share code snippets including syntax highlighting in comments and issues: [Creating Code Blocks - Syntax highlighting](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting).

One of the errors might stem of a mismatch in naming the files, e.g. the error message above states that "KatieLiu.jpg.license" is in the exclude configuration while the file on the screenshot is named "KatieLiu.jp**e**g.license". 

Also try and see if the steps lined out here ["Package resolution errors must be fixed before building - @PSchmiedmayer's Comment"](https://github.com/orgs/CS342/discussions/6#discussioncomment-4742407) can maybe help you to resolve your issue.

**Reply by katieliuu** [reply]
```swift
// swift-tools-version: 5.7
//
// This source file is part of the CS342 2023 Application project
//
// SPDX-FileCopyrightText: 2023 Stanford University
//
// SPDX-License-Identifier: MIT
//

import PackageDescription

let package = Package(
    name: "CS342MemberViews",
    platforms: [
        .iOS(.v16)
    ],
    products: [
        .library(name: "PaulSchmiedmayer", targets: ["PaulSchmiedmayer"]),
        .library(name: "OliverAalami", targets: ["OliverAalami"]),
        .library(name: "KatieLiu", targets: ["KatieLiu"])
    ],
    targets: [
        .target(
            name: "InstructorViews"
        ),
        .target(
            name: "PaulSchmiedmayer",
            dependencies: [
                .target(name: "InstructorViews")
            ],
            exclude: [
                "Resources/PaulSchmiedmayer.jpeg.license",
                "Resources/PaulSchmiedmayerBio.md.license"
            ],
            resources: [
                .process("Resources")
            ]
        ),
        .target(
            name: "OliverAalami",
            dependencies: [
                .target(name: "InstructorViews")
            ],
            exclude: [
                "Resources/OliverAalami.jpeg.license",
                "Resources/OliverAalamiBio.md.license"
            ],
            resources: [
                .process("Resources")
            ]
        ),
        .target(
            name: "KatieLiu",
            dependencies: [
                .target(name: "InstructorViews")
            ],
            exclude: [
                "Resources/KatieLiu.jpeg.license",
                "Resources/KatieLiuBio.md.license"
            ],
            resources: [
                .process("Resources")
            ]
        )
    ]
)
```

**Reply by PSchmiedmayer** [reply]
Thank you @katieliuu! 
The Package.swift file looks good to me. Did renaming the one file from jpg to jpeg and the steps lined out here ["Package resolution errors must be fixed before building - @PSchmiedmayer's Comment"](https://github.com/orgs/CS342/discussions/6#discussioncomment-4742407) solve your problem or are you still having the same warnings?

**Reply by katieliuu** [reply]
@PSchmiedmayer I still had the same issue, but after I added a .txt to each of the file names, the errors went away. Not sure why, but it seems to have fixed everything.

**Reply by PSchmiedmayer** [reply]
@katieliuu That is strange. Maybe it was really a caching error. Sometimes restarting Xcode also helps.
